### PR TITLE
feat: Mark async call chains explicitly

### PR DESCRIPTION
**Pull request by Swatinem**
## :scroll: Description

Adds the `stack_start` marker to the root frame of every chained async stack trace, allowing the UI and grouping to recognize stack trace boundaries correctly. This also simplifies the code in some ways.

## :bulb: Motivation and Context

See also https://github.com/getsentry/relay/pull/981

## :pencil: Checklist

- [ ] I reviewed the submitted code
- [x] I added tests to verify the changes
- [x] I updated the CHANGELOG if needed
- [x] I updated the docs if needed
- [ ] Review from the native team if needed
- [x] No breaking changes

### CODE REVIEW

File: Sources/Sentry/Public/SentrySDK.h
Comment (philipphofmann) [inline Sources/Sentry/Public/SentrySDK.h:29]: `l`: Do you maybe have different code formatting rules? `make format` should do the job. Can we maybe exclude these changes from the PR?

File: Sources/Sentry/SentryCrashReportConverter.m
Comment (philipphofmann) [inline Sources/Sentry/SentryCrashReportConverter.m:274]: `m`: Is there a reason why we iterate now not in reverse order anymore? Below we call `reverseObjectEnumerator` so this seems like some unnecessary work.

File: Sources/Sentry/SentryCrashReportConverter.m
Comment (philipphofmann) [inline Sources/Sentry/SentryCrashReportConverter.m:272]:
```suggestion
    SentryFrame *lastFrame = nil;
```

File: Sources/Sentry/SentryCrashReportConverter.m
Comment (philipphofmann) [inline Sources/Sentry/SentryCrashReportConverter.m:279]:
```suggestion
            if (lastFrame != null) {
```

**Review by philipphofmann — COMMENTED** [review]
A few comments. Tests would be nice before merging.

File: Sources/Sentry/Public/SentrySDK.h
Comment (Swatinem) [reply Sources/Sentry/Public/SentrySDK.h:29]: I did, and it looks like this was an automated commit: https://github.com/getsentry/sentry-cocoa/pull/1071/commits/81523f17930128e8adf970ea413a15c17ba097f4

File: Sources/Sentry/SentryCrashReportConverter.m
Comment (Swatinem) [reply Sources/Sentry/SentryCrashReportConverter.m:274]: When we hit the marker, we mark the *previous* frame, thats why we gotta iterate from top frame down.

Comment (jan-auer) [reply Sources/Sentry/SentryCrashReportConverter.m:274]: You could also just mark the next frame when iterating reverse ;)

File: Tests/SentryTests/SentryCrash/SentryStacktraceBuilderTests.swift
Comment (philipphofmann) [inline Tests/SentryTests/SentryCrash/SentryStacktraceBuilderTests.swift:96]: Thanks for adding tests 😀

File: Tests/SentryTests/Protocol/SentryFrameTests.swift
Comment (philipphofmann) [inline Tests/SentryTests/Protocol/SentryFrameTests.swift:21]: I only added this line here and reordered all the other statements so they match SentryFrame.m so I could make sure all fields are here in the tests.

File: Sources/SentryCrash/Recording/Tools/SentryCrashStackCursor.h
Comment (philipphofmann) [inline Sources/SentryCrash/Recording/Tools/SentryCrashStackCursor.h:45]: `l`: `1234` 😁. Just a coincidence or a self-chosen random value?

**Review by philipphofmann — APPROVED** [review]
From my side LGTM, would be nice to get an approval from another dev from native.

**Comment by bruno-garcia** [comment]
Given this is approved, even though we would appreciate more input from @getsentry/owners-native I'll go ahead and merge this and ship a **beta**  since we're aiming GA by next week Tuesday

**Pull request merged**
